beacon-chain/blockchain/execution_engine.go

```diff
@@ -261,10 +261,13 @@ func (s *Service) getPayloadAttribute(ctx context.Context, st state.BeaconState,
 
 	// Get previous randao.
 	st = st.Copy()
-	st, err := transition.ProcessSlotsIfPossible(ctx, st, slot)
-	if err != nil {
-		log.WithError(err).Error("Could not process slots to get payload attribute")
-		return false, emptyAttri, 0
+	if slot > st.Slot() {
+		var err error
+		st, err = transition.ProcessSlotsUsingNextSlotCache(ctx, st, headRoot, slot)
```

Comment:
This doesnt seem right. The method signature for ProcessSlotsUsingNextSlotCache requires the parent root in the 3rd argument, not the head root? If that is the case, why not read the parent root of `st`?

https://github.com/prysmaticlabs/prysm/blob/fa7a2bd44589a4c9861deff3d5157fe0f03bb0d9/beacon-chain/core/transition/transition.go#L142-L146

Reply:
ah nvm -- it calls for the parent state too